conformance/lang/values-types-and-variables/nil-literal.balt

```diff
@@ -0,0 +1,173 @@
+Test-Case: output
+Description: Test nil type descriptor with nil literal '()'.
+Labels: nil-type-descriptor, nil-literal
+
+public function main() {
+    () x = ();
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test nil type descriptor with nil literal 'null'.
+Labels: nil-type-descriptor, nil-literal
+
+public function main() {
+    () x = null;
+    io:println(x.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test value and assignability of the nil literal.
+Labels: nil-type-descriptor, nil-literal, equality-expr
+
+public function main() {
+    () x = ();
+    () y = null;
+    io:println(x == y); // @output true
```

Comment:
This is part of the nil type, Since are testing the nil literal here, we can write this as `() == null`. Also we can include `!=` `===` and `!==` with nil literal. `is` expression another place we can use nil literal. Please add a test case to cover that.